### DISCUSSION TITLE
How to build wheel compatible with multiple python versions

### DISCUSSION THREAD

**Discussion by qqlearn123** [PyO3/maturin#1630] [Q&A] [answered]
I have a mixed rust/python project and I built a `wheel` using `maturin build`. The `whl` file I got is named something like `cp39-cp39-manylinux_2_17_x86_64...` since I used `python3.9` to build it. Later, I found out this `wheel` is only installable via `pip install` for `python3.9` in particular.

How can I build a `whl` file compatible with multiple python versions using just one python version to build (say, build with python3.9, and the resulting will be compatible with python3.9 and up)?

I saw the maturin `whl` file on the public PYPI site is called something like `py3-none-manylinux...`. So, I think there must be a way to do it.

**Comment by konstin** [comment, marked as answer]
You need to use the `abi3` feature from pyo3: https://pyo3.rs/v0.18.3/features.html#abi3
